server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -70,36 +69,58 @@ class InteractiveSession(
     val conf = prepareConf(request.conf, request.jars, request.files, request.archives,
       request.pyFiles)
 
-    info(s"Creating LivyClient for sessionId: $id")
-    val builder = new LivyClientBuilder()
-      .setConf("spark.app.name", s"livy-session-$id")
-      .setAll(conf.asJava)
-      .setConf("livy.client.sessionId", id.toString)
-      .setConf(RSCConf.Entry.DRIVER_CLASS.key(), "com.cloudera.livy.repl.ReplDriver")
-      .setURI(new URI("rsc:/"))
+    val builderProperties = mutable.Map[String, String]()
+    builderProperties ++= conf
 
     def mergeConfList(list: Seq[String], key: String): Unit = {
       if (list.nonEmpty) {
-        val newList = (list ++ conf.get(key)).mkString(",")
-        builder.setConf(key, newList)
+        builderProperties.get(key) match {
+          case None =>
+            builderProperties.put(key, list.mkString(","))
+          case Some(oldList) =>
+            val newList = (oldList :: list.toList).mkString(",")
+            builderProperties.put(key, newList)
+        }
       }
     }
 
+    def mergeHiveSiteAndHiveDeps(): Unit = {
+      val sparkFiles = conf.get("spark.files").map(_.split(",")).getOrElse(Array.empty[String])
+      if (livyConf.getBoolean(LivyConf.ENABLE_HIVE_CONTEXT)) {
+        hiveSiteFile(sparkFiles, livyConf) match {
+          case (_, true) =>
+            info("Enable HiveContext because hive-site.xml is found in user request.")
+            mergeConfList(datanucleusJars(livyConf), LivyConf.SPARK_JARS)
+          case (Some(file), false) =>
+            info("Enable HiveContext because hive-site.xml is found under classpath, "
+              + file.getAbsolutePath)
+            mergeConfList(List(file.getAbsolutePath), LivyConf.SPARK_FILES)
+            mergeConfList(datanucleusJars(livyConf), LivyConf.SPARK_JARS)
+          case (None, false) =>
+            warn("Enable HiveContext but no hive-site.xml found under" +
```

Comment:
I don't think this warrants a warning. If you're gonna allow hdfs-site in the user request, it's perfectly normal for an admin to not provide their own default, and require users to provide their own if they want to use HiveContext. The code in the interpreter should detect that and not start a HiveContext.

So, like the others, `debug`.